util/resource/resource.go

```diff
@@ -17,61 +17,54 @@ limitations under the License.
 // Package resource implements resource utilites.
 package resource
 
-import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+import (
+	"sort"
 
-var defaultCreatePriorities = []string{
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var priorityMap = map[string]int{
 	// Namespaces go first because all namespaced resources depend on them.
-	"Namespace",
+	"Namespace": -100,
 	// Custom Resource Definitions come before Custom Resource so that they can be
 	// restored with their corresponding CRD.
-	"CustomResourceDefinition",
+	"CustomResourceDefinition": -99,
 	// Storage Classes are needed to create PVs and PVCs correctly.
-	"StorageClass",
+	"StorageClass": -98,
 	// PVs go before PVCs because PVCs depend on them.
-	"PersistentVolume",
+	"PersistentVolume": -97,
 	// PVCs go before pods or controllers so they can be mounted as volumes.
-	"PersistentVolumeClaim",
+	"PersistentVolumeClaim": -96,
 	// Secrets and ConfigMaps go before pods or controllers so they can be mounted as volumes.
-	"Secret",
-	"ConfigMap",
+	"Secret":    -95,
+	"ConfigMap": -94,
 	// Service accounts go before pods or controllers so pods can use them.
-	"ServiceAccount",
+	"ServiceAccount": -93,
 	// Limit ranges go before pods or controllers so pods can use them.
-	"LimitRange",
+	"LimitRange": -92,
 	// Pods go before ReplicaSets
-	"Pods",
+	"Pods": -91,
 	// ReplicaSets go before Deployments
-	"ReplicaSet",
+	"ReplicaSet": -90,
 	// Endpoints go before Services
-	"Endpoints",
+	"Endpoints": -89,
+}
+
+// PriorityLess returns true if o1 should be created before o2.
+// To be used in sort.{Slice,SliceStable} to sort manifests in place.
+func PriorityLess(o1, o2 unstructured.Unstructured) bool {
```

Comment:
nit: any reason for this to be exported?

Reply:
It was originally exported, so that we could use it with `sort.Slice` across the project... but since we're not doing that, I actually [suggested](https://github.com/kubernetes-sigs/cluster-api/pull/9251#issuecomment-1686151711) we can unexport it to make the API cleaner.

Reply:
Unexported.